integration/spark-common/src/main/scala/org/apache/carbondata/spark/rdd/CarbonScanRDD.scala

```
@@ -231,4 +248,21 @@ class CarbonScanRDD[V: ClassTag](
     val firstOptionLocation = theSplit.split.value.getLocations.filter(_ != "localhost")
     firstOptionLocation
   }
+
+  def createVectorizedCarbonRecordReader(queryModel: QueryModel): RecordReader[Void, Object] = {
+    val name = "org.apache.carbondata.spark.vectorreader.VectorizedCarbonRecordReader"
+    try {
+      val cons = Class.forName(name).getDeclaredConstructors
+      cons.head.setAccessible(true)
+      cons.head.newInstance(queryModel).asInstanceOf[RecordReader[Void, Object]]
+    } catch {
+      case e: Exception =>
+        null
```

should not ignore exception